### ISSUE TITLE
[BUG] Control+Shift+ArrowLeft fails on contenteditables with FF + Linux

### ISSUE THREAD

**Issue by trueadm** [microsoft/playwright#5929]
**Context:**
- Playwright Version: 1.9.2
- Operating System: Linux
- Node.js version: 12
- Browser: Firefox

**Code Snippet**

```javascript
const {chromium, webkit, firefox} = require('playwright');

(async () => {
  const browser = await chromium.launch();
  // ...
  // locate <div contenteditable="true" id="contenteditable"></div>
  await page.focus('#contenteditable');
  await page.keyboard.type('Hello world');
  await page.keyboard.down('Control');
  await page.keyboard.down('Shift');
  await page.keyboard.press('ArrowLeft');
  await page.keyboard.up('Shift');
  await page.keyboard.up('Control');
  // Expectation is that the word "world" should be selected.
})();
```

**Describe the bug**

This was a strange issue for us. We encountered it on our CI that runs Linux and after playing around, we were able to determine that this only occurs for Firefox on Linux. Specifically, it also only seems to affect `contenteditable="true"` nodes. It works fine for `<input>` elements.

It seems that this does nothing in FF for Linux. The expected behavior is to select the word "world", as the keyboard shortcut Control + Shift + ArrowLeft is intended to do (works fine natively using Linux and contenteditables with FF).

Note: when we run this test, we are also running the same test on other browsers too. Could it be that there's a foreground issue with selection and FF?

**Comment by JoelEinbinder** [comment]
Added a test that passes for me on firefox/linux. Maybe there is something mores specific happening with your contenteditable div?

**Comment by trueadm** [comment]
@JoelEinbinder Interesting. I'll dig more into it today. I suspect it's a foreground problem maybe on the CI machine.

**Comment by trueadm** [comment]
I'm fairly certain something is going wrong here. I've stripped back the test case so we aren't running any JS and the page is just an empty document with a contenteditable div. This is the HTML:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
  </head>
  <body>
    <div id="contenteditable" contenteditable="true"></div>
  </body>
</html>

```

This is the test case:

```js
it(`Can select and delete a word`, async () => {
  const {page} = e2e;

  await page.focus('#contenteditable');
  await page.keyboard.type('Hello world');
  await page.keyboard.down('Control');
  await page.keyboard.down('Shift');
  await page.keyboard.press('ArrowLeft');
  await page.keyboard.up('Shift');
  await page.keyboard.up('Control');
  // Log screenshot

  await page.keyboard.press('Backspace');
  const remainingText = await page.textContent(
    '#contenteditable',
  );
  expect(remainingText).toBe('Hello ');
});
```

The screenshot just before backspace in Linux + FF on CI:

![cbimage](https://user-images.githubusercontent.com/1519870/112361413-42173900-8ccb-11eb-89be-4ade3f0fcb56.jpg)

Here's out Github Actions config:

```yml
name: Tests
on: [push]
jobs:
  e2e:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [14.x]
    steps:
    - uses: actions/checkout@v2
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}
    - run: yarn install
    - run: yarn build
    - run: yarn test
      env:
        CI: true
```

**Comment by trueadm** [comment]
Interestingly, if I change the test case to be:

```js
await page.focus('#contenteditable');
await page.keyboard.type('Hello world');
await page.keyboard.down('Shift');
await page.keyboard.press('ArrowLeft');
await page.keyboard.press('ArrowLeft');
await page.keyboard.press('ArrowLeft');
await page.keyboard.press('ArrowLeft');
await page.keyboard.press('ArrowLeft');
await page.keyboard.up('Shift');
```

It works fine on FF+Linux on CI. So it does seem like a bug with Playwright.

**Comment by JoelEinbinder** [comment]
I can repro it on 1.9.2 but not on tip of tree. Looks like it was fixed via the latest firefox roll. v1.10 should be coming shortly with the fix. Or you can install npm install playwright@next if you are impatient.

**Comment by trueadm** [comment]
@JoelEinbinder Awesome! Thanks :)

**Comment by trueadm** [comment]
This is so strange. With playwright@next, FF now passes for Linux on CI but fails when run locally on my Mac.

**Comment by JoelEinbinder** [comment]
Are you swapping ctrl for alt to get the correct shortcut on your mac?

**Comment by trueadm** [comment]
@JoelEinbinder Yep, we use a function that detects platform and uses the right key.

**Comment by JoelEinbinder** [comment]
Fixed via #6052 verified via #6034
